app/imageswap/imageswap.py

```diff
@@ -82,34 +83,46 @@ def mutate():
+    # Skip patching if disable label is found and set to "disable"
+    if (
+        "labels" in workload_metadata
+        and imageswap_disable_label in workload_metadata["labels"]
+        and workload_metadata["labels"][imageswap_disable_label] == "disabled"
+    ):
 
-        for container_spec in modified_spec["request"]["object"]["spec"]["containers"]:
+        app.logger.info(f'Disable label "{imageswap_disable_label}=disabled" detected, skipping image swap.')
+        needs_patch = False
 
-            app.logger.info(f"Processing container: {namespace}/{workload}")
-            needs_patch = swap_image(container_spec)
+    else:
 
-        if "initContainers" in modified_spec["request"]["object"]["spec"]:
+        # Change workflow/json path based on K8s object type
+        if workload_type == "Pod":
 
-            for init_container_spec in modified_spec["request"]["object"]["spec"]["initContainers"]:
+            for container_spec in modified_spec["request"]["object"]["spec"]["containers"]:
 
-                app.logger.info(f"Processing init-container: {namespace}/{workload}")
-                needs_patch = swap_image(init_container_spec)
+                app.logger.info(f"Processing container: {namespace}/{workload}")
+                needs_patch = swap_image(container_spec)
 
-    else:
+            if "initContainers" in modified_spec["request"]["object"]["spec"]:
+
+                for init_container_spec in modified_spec["request"]["object"]["spec"]["initContainers"]:
+
+                    app.logger.info(f"Processing init-container: {namespace}/{workload}")
+                    needs_patch = swap_image(init_container_spec)
+
+        else:
 
-        for container_spec in modified_spec["request"]["object"]["spec"]["template"]["spec"]["containers"]:
+            for container_spec in modified_spec["request"]["object"]["spec"]["template"]["spec"]["containers"]:
 
-            app.logger.info(f"Processing container: {namespace}/{workload}")
-            needs_patch = swap_image(container_spec)
+                app.logger.info(f"Processing container: {namespace}/{workload}")
+                needs_patch = swap_image(container_spec)
 
-        if "initContainers" in modified_spec["request"]["object"]["spec"]["template"]["spec"]:
+            if "initContainers" in modified_spec["request"]["object"]["spec"]["template"]["spec"]:
 
-            for init_container_spec in modified_spec["request"]["object"]["spec"]["template"]["spec"]["initContainers"]:
+                for init_container_spec in modified_spec["request"]["object"]["spec"]["template"]["spec"]["initContainers"]:
 
-                app.logger.info(f"Processing init-container: {namespace}/{workload}")
-                needs_patch = swap_image(init_container_spec)
+                    app.logger.info(f"Processing init-container: {namespace}/{workload}")
+                    needs_patch = swap_image(init_container_spec)
 
     if needs_patch:
```

Comment:
Not sure this is germane strictly speaking but in trying to understand the overall logic I ran across this: Why does this log "Doesn't need patch" when it's in the `if needs_patch:` block? It looks like the remainder of the block's logic does create a patch and embed the patch inside the admission_response.

Reply:
Ok, I think this was just a latest copy/paste error from some point and/or things got out of place when I moved things around for the disable label logic. Not sure...but I pushed a fix that should resolve it.